File: Frontend/src/types/analytics.ts
import { Question } from './question';
import { RoomQuestion, RoomQuestionEvaluation } from './room';
import { User, UserType } from './user';

export interface AnalyticsQuestionsUserReactionSummary {
  id: string;
  type: string;
  count: number;
}

export interface AnalyticsQuestionsUserReaction {
  id: string;
  type: string;
}

export interface AnalyticsQuestionsExpert extends User {
  nickname: string;
  reactionsSummary: AnalyticsQuestionsUserReactionSummary[];
}

interface AnalyticsQuestionsViwer {
  reactionsSummary: AnalyticsQuestionsUserReactionSummary[];
}

export interface AnalyticsSummaryQuestions extends Question {
  status: string;
  viewers: AnalyticsQuestionsViwer[] | null;
  experts: AnalyticsQuestionsExpert[] | null;
}

export interface AnalyticsSummary {
  questions: AnalyticsSummaryQuestions[];
}

export interface AnalyticsQuestions extends RoomQuestion {
  averageMark: number;
  users: Array<{
    id: User['id'];
    evaluation?: Omit<RoomQuestionEvaluation, 'id'>;
  }>;
}

export interface AnalyticsUserReview {
  userId: User['id'];
  nickname: User['nickname'];
  avatar?: User['avatar'];
  participantType: UserType;
  averageMark?: number;
  comment: string;
}

export interface Analytics {
  completed: boolean;
  averageMark?: number;
  questions: AnalyticsQuestions[];
  userReview: AnalyticsUserReview[];
}

File: Backend/Interview.Domain/Rooms/Analytics.cs
namespace Interview.Domain.Rooms;

public class Analytics
{
    public List<AnalyticsQuestion>? Questions { get; set; }

    public class AnalyticsQuestion
    {
        public Guid Id { get; set; }

        public string Value { get; set; } = string.Empty;

        public string Status { get; set; } = string.Empty;

        public List<AnalyticsUser>? Users { get; set; }
    }

    public class AnalyticsReaction
    {
        public Guid Id { get; set; }

        public string Type { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; }
    }

    public class AnalyticsReactionSummary
    {
        public Guid Id { get; set; }

        public string Type { get; set; } = string.Empty;

        public int Count { get; set; }
    }

    public class AnalyticsUser
    {
        public Guid Id { get; set; }

        public string Nickname { get; set; } = string.Empty;

        public string Avatar { get; set; } = string.Empty;

        public string ParticipantType { get; set; } = string.Empty;
    }
}
